controllers/openstackdataplanenode_controller.go

```diff
@@ -118,15 +129,25 @@ func (r *OpenStackDataPlaneNodeReconciler) Provision(ctx context.Context, instan
 }
 
 // GenerateInventory yields a parsed Inventory
-func (r *OpenStackDataPlaneNodeReconciler) GenerateInventory(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlaneNode) error {
+func (r *OpenStackDataPlaneNodeReconciler) GenerateInventory(ctx context.Context, instance *dataplanev1beta1.OpenStackDataPlaneNode, instanceRole *dataplanev1beta1.OpenStackDataPlaneRole) error {
```

Comment:
wrt batching/aggregating inventories for future use - we might also want to put each node into a children  group defined from its parent role CR's metadata name.

then, after individual inventories for DP Node CRs are created, we can leverage dynamic merging feature for AEE. For example, there could be a volumeMount for each inventory file in AEE job pod aggregated in the  '/inventories' directory. Then it can be batched just as running ansible with -i /inventories

Reply:
> wrt batching/aggregating inventories for future use - we might also want to put each node into a children group defined from its parent role CR's metadata name.

So right now there's one inventory created per node. To do that we'd need an inventory created per role. I like this idea, though I think we should do it in a separate PR after we have the Ansible execution working with 1 node to 1 inventory. 

> then, after individual inventories for DP Node CRs are created, we can leverage dynamic merging feature for AEE. For example, there could be a volumeMount for each inventory file in AEE job pod aggregated in the '/inventories' directory. Then it can be batched just as running ansible with -i /inventories

Yeah, I think that would good; something to do in a future PR.

Reply:
I meant to say we can still create a single node inventories, just adding each node there into its corresponding group. Not combining all nodes of a role into inventory file - but rely on automatic merging it by ansible

Reply:
> I meant to say we can still create a single node inventories, just adding each node there into its corresponding role group. Not combining all nodes of a role into inventory file - but rely on automatic merging it by ansible

So when I add one node (which creates an inventory) I also add an inventory group based on it's role. If N nodes are added then we have N inventory config-maps. One Ansible runner could then mounts all N inventories in a directory and the target it of the run could be all members of that group. Ansible will then merge the inventories because of the way `ansible -i /path/to/dir` behaves.